src/crystal/system/unix/mime.cr

```diff
@@ -0,0 +1,24 @@
+module Crystal::System::MIME
+  MIME_SOURCES = [
+    "/etc/mime.types",
+    "/etc/apache2/mime.types",
+    "/etc/apache/mime.types",
+  ]
+
+  {% if flag?(:freebsd) %}
+    MIME_SOURCES << "/usr/local/etc/mime.types"
+  {% elsif flag?(:openbsd) %}
+    MIME_SOURCES << "/usr/share/misc/mime.types"
+  {% end %}
+
+  # Load MIME types from operating system source.
+  def self.load
+    MIME_SOURCES.each do |path|
+      next unless ::File.exists?(path)
+      ::File.open(path) do |file|
+        ::MIME.load_mime_database file
```

Comment:
That should work the other way around: `Crystal::System::MIME` should provide a method that returns (or yields) a `mime.types` file that `MIME` would try to parse.

That would avoid the circular dependency that `MIME` depends on `Crystal::System::MIME` which itself depends on `MIME`.

Reply:
I don't like the circular dependency either. But the fact that `Crystal::System::MIME.load` utilizes `MIME.load_mime_database` is already a platform specific implementation. 

It could certainly be reversed if `load` yields if it needs to load a file. On windows it just wouldn't yield, but load the database differently. But that's baking platform-specifics into the abstraction API.

That's how I chose to implement it  and I think it should stay this way. But I won't hurt about changing if requested.

Reply:
It's similar with `Crystal::System::Time.load_localtime` btw., the Unix implementation calls the non-platform-specific implementation `Time::Location.read_zoneinfo`.

Reply:
I think what makes me wary is that something from Crystal::System is initializing an external namespace; and ::MIME should initialize itself using Crystal::System. I wish we could find a better way.

Reply:
We could pull the platform-dependent initialization implementations out of `Crystal::System` into the main namespace. But that doesn't change anything, really.